aviatrix/resource_aviatrix_account.go

```diff
@@ -387,6 +388,16 @@ func resourceAviatrixAccount() *schema.Resource {
 				Computed:    true,
 				Description: "AWS Secret Region CAP Certificate Key file path on the controller.",
 			},
+			"group_names": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					groupNameOld, _ := d.GetChange("group_names")
+					return len(groupNameOld.([]interface{})) != 0
```

Comment:
The `old` and `new` parameters passed into the DiffSuppressFunc should already be the values that `d.GetChange` returns. You shouldn't need to call that function.

Also, can you list which cases you tested so that we can be sure the DIffSuppressFunc works as intended.

Reply:
Also did you test what happens when the RBAC groups are changed outside this resource? Does the return value from the API actually change?

Reply:
yes, I have tested:
first try to create an Access account: "access-account1", give this account with two RBAC group: ["group-a", "group-b"], then, when we delete one group from UI, and terraform apply again, it will show "No changes."
second test is when we use the resource: "aviatrix_rbac_group_access_account_attachment" to add a RBAC group: "group-c" to the Access account: "access-account1", and then terraform apply again, it shows "No changes".

I will add this test case in the JIRA ticket:
# Create an Aviatrix AWS Account with IAM roles
resource "aviatrix_account" "temp_acc_aws" {
  account_name       = "access-account1"
  cloud_type         = 1
  aws_account_number = "722837227438"
  aws_iam            = true
  aws_role_app       = "arn:aws:iam::722837227438:role/aviatrix-role-app"
  aws_role_ec2       = "arn:aws:iam::722837227438:role/aviatrix-role-ec2"
  group_names        = ["group-a", "group-b"]
}

# Create an Aviatrix RBAC Group Access Account Attachment
resource "aviatrix_rbac_group_access_account_attachment" "test_attachment1" {
  group_name          = "group-c"
  access_account_name = "access-account1"
}

Reply:
> The `old` and `new` parameters passed into the DiffSuppressFunc should already be the values that `d.GetChange` returns. You shouldn't need to call that function.
> 
> Also, can you list which cases you tested so that we can be sure the DIffSuppressFunc works as intended. This list can just be added as a comment to the JIRA ticket.

The `old` and `new` may generate undefined behaviors here because this attribute is a list.